pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -188,15 +197,24 @@ func (r *SubnetPortReconciler) deleteSubnetPortByName(ctx context.Context, ns st
 		return err
 	}
 
+	var hasSubnetPort bool
+	var externalIpAddress *string
 	for _, nsxSubnetPort := range nsxSubnetPorts {
 		if crSubnetPortIDsSet.Has(*nsxSubnetPort.Id) {
 			log.Info("skipping deletion, SubnetPort CR still exists in K8s", "ID", *nsxSubnetPort.Id)
+			hasSubnetPort = true
 			continue
 		}
+		if nsxSubnetPort.ExternalAddressBinding != nil && nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress != nil && *nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress != "" {
+			externalIpAddress = nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress
+		}
 		if err := r.SubnetPortService.DeleteSubnetPort(nsxSubnetPort); err != nil {
 			return err
 		}
 	}
+	if !hasSubnetPort && externalIpAddress != nil {
+		r.collectAddressBindingGarbage(ctx, &ns, externalIpAddress)
```

Comment:
`collectAddressBindingGarbage(ctx context.Context, namespace, ipAddress *string)`
namespace and ipAddress is *string since we'll set it to nil when we want to do full GC, and set it to valid string to do a partial GC for here.